scripts/west_commands/runners/pyocd.py

```diff
@@ -38,8 +39,8 @@ def __init__(self, cfg, target,
         ''' if running in WSL, call py.exe to run pyOCD in Windows '''
         ''' most debuggers are not accessible directly in WSL, but '''
         ''' the pyOCD server is still accessible to gdb inside WSL '''
-        self.pyocd = [wsl_py_path, '-m', pyocd] if wsl_py_path is not None else [pyocd]
-        self.wsl_py_path = wsl_py_path
+        self.on_wsl = 'Microsoft' in uname().release
```

Comment:
Please:

1. Squash these changes into the previous commit
2. Defer the work from the constructor to the do_run() call

Note that `uname()` is a syscall (even though its results are cached by the python standard library). The runners are generally structured to avoid 'doing work' in the constructor because the error handling in the higher layers is focused on when the do_run() call is made. Instead of the way it's done here, please just save the user arguments in self, and decide at do_run() time whether we're on WSL and how to handle that.

Reply:
I moved the uname check and pushed it.  I'll squash these commits tomorrow.

Reply:
I squashed the commits, but now my branch is diverged.  Is it safe to force push?

Reply:
> I squashed the commits

Thanks

> but now my branch is diverged. Is it safe to force push?

Yes. Please read https://docs.zephyrproject.org/latest/contribute/index.html#contribution-workflow for full details.